qiskit/visualization/state_visualization.py

```diff
@@ -540,55 +538,89 @@ def plot_state_qsphere(rho, figsize=None):
             ax.w_xaxis.line.set_color((1.0, 1.0, 1.0, 0.0))
             ax.w_yaxis.line.set_color((1.0, 1.0, 1.0, 0.0))
             ax.w_zaxis.line.set_color((1.0, 1.0, 1.0, 0.0))
+
             # Get rid of the ticks
             ax.set_xticks([])
             ax.set_yticks([])
             ax.set_zticks([])
 
             d = num
-            for i in range(2**num):
+            for i in range(2 ** num):
+
                 # get x,y,z points
                 element = bin(i)[2:].zfill(num)
                 weight = element.count("1")
                 zvalue = -2 * weight / d + 1
                 number_of_divisions = n_choose_k(d, weight)
                 weight_order = bit_string_index(element)
-                # if weight_order >= number_of_divisions / 2:
-                #    com_key = compliment(element)
-                #    weight_order_temp = bit_string_index(com_key)
-                #    weight_order = np.floor(
-                #        number_of_divisions / 2) + weight_order_temp + 1
-                angle = weight_order * 2 * np.pi / number_of_divisions
-                xvalue = np.sqrt(1 - zvalue**2) * np.cos(angle)
-                yvalue = np.sqrt(1 - zvalue**2) * np.sin(angle)
-                ax.plot([xvalue], [yvalue], [zvalue],
-                        markerfacecolor=(.5, .5, .5),
-                        markeredgecolor=(.5, .5, .5),
-                        marker='o', markersize=10, alpha=1)
+                angle = (float(weight) / d) * (np.pi * 2) + (weight_order * 2 * (np.pi / number_of_divisions))
+
+                if (weight > d / 2) or (((weight == d / 2) and (weight_order >= number_of_divisions / 2))):
```

Comment:
This one also needs to be under 100 char.

Reply:
Addressed